### PR TITLE
add prefill transform for 1990 email and phone numbers

### PR DESCRIPTION
**Pull request by rianfowler**
This can be test locally against the `prefill-test` branch of `vets-api`. All logged in users will return the prefill data for contact info.

### CODE REVIEW

File: src/applications/edu-benefits/1990/helpers.jsx [hunk 1]
```diff
@@ -2,6 +2,51 @@ import React from 'react';
...
 import { transformForSubmit } from 'us-forms-system/lib/js/helpers';
 
+export function prefillTransformer(pages, formData, metadata) {
+
+  let newContactInfo = {};
+
+  if (formData && formData.email) {
+    newContactInfo = {
+      email: formData.email,
+      'view:confirmEmail': formData.email
```
Comment (jbalboni) [inline src/applications/edu-benefits/1990/helpers.jsx]: I wonder if we should make them type this?

File: src/applications/edu-benefits/1990/helpers.jsx [hunk 2]
```diff
@@ -2,6 +2,51 @@ import React from 'react';
...
+    };
+  }
+
+  if (formData && formData.homePhone) {
+    newContactInfo = {
+      ...newContactInfo,
```
Comment (jbalboni) [inline src/applications/edu-benefits/1990/helpers.jsx]: You could set `homePhone` on `newContactInfo` directly. You created it in this function, so mutating it won't cause any issues.

File: src/applications/edu-benefits/1990/helpers.jsx [hunk 3]
```diff
@@ -2,6 +2,51 @@ import React from 'react';
...
+      homePhone: formData.homePhone
+    };
+  }
+
+  if (formData && formData.mobilePhone) {
```
Comment (jbalboni) [inline src/applications/edu-benefits/1990/helpers.jsx]: You also should remove these contact fields from the top level of the form data object you return. I think they'll get overwritten on submit, but we shouldn't leave them hanging around.

File: src/applications/edu-benefits/1990/helpers.jsx (hunk 1 above)
Comment (rianfowler) [reply src/applications/edu-benefits/1990/helpers.jsx]: Good idea- I'll take this part out.

File: src/applications/edu-benefits/1990/helpers.jsx [hunk 4]
```diff
@@ -1,7 +1,44 @@
...
+  };
+
+  if (formData) {
+    const {
+      email,
+      homePhone,
+      mobilePhone
+    } = formData;
+
+    delete newFormData.email;
```
Comment (rianfowler) [inline src/applications/edu-benefits/1990/helpers.jsx]: Remove these from `formData`.

File: src/applications/edu-benefits/1990/helpers.jsx [hunk 5]
```diff
@@ -1,7 +1,44 @@
...
+    delete newFormData.homePhone;
+    delete newFormData.mobilePhone;
+
+    const newContactInfo = {
+      ...(email && { email }),
```
Comment (rianfowler) [inline src/applications/edu-benefits/1990/helpers.jsx]: consolidated these into a single statement. seems easier to read

File: src/applications/edu-benefits/1990/helpers.jsx [hunk 6]
```diff
@@ -1,7 +1,43 @@
...
+      ...(homePhone && { homePhone }),
+      ...(mobilePhone && { mobilePhone })
+    };
+
+    if (!_.isEmpty(newContactInfo)) {
```
Comment (rianfowler) [inline src/applications/edu-benefits/1990/helpers.jsx]: seemed clearer to use a lodash function here.

File: src/applications/edu-benefits/1990/helpers.jsx (hunk 5 above)
Comment (rianfowler) [reply src/applications/edu-benefits/1990/helpers.jsx]: JS should have a better way of doing this kind of thing.

**Comment by rianfowler** [comment]
@jbalboni addressed comments- thanks for the feedback.

**Review by jbalboni — APPROVED** [review]

**Pull request merged**